### ISSUE TITLE
Hardnegative mining problem for SSD with pose estimation 

### ISSUE THREAD

**Issue by h19920918** [weiliu89/caffe#742]
Sorry for write in here but I don't have a choice.

Thank you for your development.
I found you are related to ssd with pose estimation that the paper name is 'Fast Single Shot Detection and Pose Estimation'.

It doesn't exist any code in website, so, our team is making your model relied on the paper.
Our development is almost done, but there is a problem during training.

The main problem is when we made the hard-negative mining that you written, but pose layer didn't converge in contrast to confidence and localization.
We made hard-negative mining for pose layer through taking same way in SSD confidence layer.
The paper didn't mention hard-negative mining for pose layer just mentioned following SSD training way.

Please, give to me some advice.

I also upload our training log below,

I0918 21:35:02.701439 23957 multibox_pose_loss_layer.cpp:335] loc_loss : 64.7937
I0918 21:35:02.701480 23957 multibox_pose_loss_layer.cpp:336]  loc_loss_.cpu_data()[0] / normalizer : **1.50683**
I0918 21:35:02.701514 23957 multibox_pose_loss_layer.cpp:342] conf_loss : 111.588
I0918 21:35:02.701550 23957 multibox_pose_loss_layer.cpp:343] conf_loss_.cpu_data()[0] / normalizer : **2.59508**
I0918 21:35:02.701629 23957 multibox_pose_loss_layer.cpp:349]  pose_loss : 853.808
I0918 21:35:02.701665 23957 multibox_pose_loss_layer.cpp:350]  pose_loss_.cpu_data()[0] / normalizer : **19.856**
I0918 21:35:07.049504 24031 multibox_pose_loss_layer.cpp:334]  normlizer : 30
I0918 21:35:07.049572 24031 multibox_pose_loss_layer.cpp:335]  loc_loss : 84.366
I0918 21:35:07.049590 24031 multibox_pose_loss_layer.cpp:336]  loc_loss_.cpu_data()[0] / normalizer : **2.8122**
I0918 21:35:07.049607 24031 multibox_pose_loss_layer.cpp:342]  conf_loss : 96.8588
I0918 21:35:07.049619 24031 multibox_pose_loss_layer.cpp:343]  conf_loss_.cpu_data()[0] / normalizer : **3.22863**
I0918 21:35:07.049631 24031 multibox_pose_loss_layer.cpp:349]  pose_loss : 544.698
I0918 21:35:07.049644 24031 multibox_pose_loss_layer.cpp:350]  pose_loss_.cpu_data()[0] / normalizer : **18.1566**
I0918 21:35:07.103458 23957 multibox_pose_loss_layer.cpp:334]  normlizer : 39
I0918 21:35:07.103515 23957 multibox_pose_loss_layer.cpp:335]  loc_loss : 29.3335
I0918 21:35:07.103530 23957 multibox_pose_loss_layer.cpp:336]  loc_loss_.cpu_data()[0] / normalizer : **0.752141**
I0918 21:35:07.103543 23957 multibox_pose_loss_layer.cpp:342]  conf_loss : 83.967
I0918 21:35:07.103555 23957 multibox_pose_loss_layer.cpp:343]  conf_loss_.cpu_data()[0] / normalizer : **2.153**
I0918 21:35:07.103567 23957 multibox_pose_loss_layer.cpp:349]  pose_loss : 820.878
I0918 21:35:07.103580 23957 multibox_pose_loss_layer.cpp:350]  pose_loss_.cpu_data()[0] / normalizer : **21.0482**

As you saw, confidence and localization layers did converge well, but not pose layer.
The result is after 2000 iteration, we used pretrained model(PASCAL VOC + COCO) provided by you and lr = 0.0001.
When we used lr = 0.001, localization layer became nan.

**Comment by h19920918** [comment]
@weiliu89 Please, watch and advice to me.. It doesn't exist anywhere except you.
